crates/sui-core/src/jsonrpc_index.rs

```diff
@@ -451,7 +452,7 @@ impl IndexStore {
     }
 
     #[instrument(skip_all)]
-    pub async fn index_coin(
+    pub fn index_coin(
```

Comment:
once jsonrpc goes away we'll also be able to remove this type of indexing from the critical path